packages/aws-cdk-lib/aws-stepfunctions/lib/states/distributed-map.ts

```diff
@@ -0,0 +1,770 @@
+   * @returns - JSON object
+   */
+  public render(): any {
+    return FieldUtils.renderObject({
+      Resource: this.resource,
+      ReaderConfig: {
+        InputType: this.inputType,
+        ...(this.maxItems && { MaxItems: this.maxItems }),
+      },
+      Parameters: {
+        Bucket: this.bucket.bucketName,
+        Key: this.key,
+      },
+    });
+  }
+
+  /**
+   * Compile policy statements to provide relevent permissions to the state machine
+   */
+  public providePolicyStatements(): iam.PolicyStatement[] {
+    const resource = `arn:${Aws.PARTITION}:s3:::${this.bucket.bucketName}/*`;
+
+    return [
+      new iam.PolicyStatement({
+        actions: [
+          's3:GetObject',
+        ],
+        resources: [resource],
+      }),
+    ];
+  }
+}
+
+/**
+ * Item Reader configuration for iterating over items in a JSON array stored in a S3 file
+ */
+export class S3JsonItemReader extends S3ItemReader {
+  protected readonly inputType: string = 'JSON';
+}
+
+/**
+ * CSV header location options
+ */
+export enum CsvHeaderLocation {
+  /**
+   * Headers will be read from first row of CSV file
+   */
+  FIRST_ROW = 'FIRST_ROW',
+
+  /**
+   * Headers are provided in CSVHeaders property
+   */
+  GIVEN = 'GIVEN',
+}
+
+/**
+ * Configuration for CSV header options for a CSV Item Reader
+ */
+export class CsvHeaders {
+  /**
+   * Configures S3CsvItemReader to read headers from the first row of the CSV file
+   * @returns - CsvHeaders
+   */
+  public static useFirstRow(): CsvHeaders {
+    return new CsvHeaders(CsvHeaderLocation.FIRST_ROW);
+  }
+
+  /**
+   * Configures S3CsvItemReader to use the headers provided in the `headers` parameter
+   * @param headers - List of headers
+   * @returns - CsvHeaders
+   */
+  public static use(headers: string[]): CsvHeaders {
+    return new CsvHeaders(CsvHeaderLocation.GIVEN, headers);
+  }
+
+  /**
+   * Location of headers in CSV file
+   */
+  public readonly headerLocation: CsvHeaderLocation;
+
+  /**
+   * List of headers if `headerLocation` is `GIVEN`
+   */
+  public readonly headers?: string[];
+
+  private constructor(headerLocation: CsvHeaderLocation, headers?: string[]) {
+    this.headerLocation = headerLocation;
+    this.headers = headers;
+  }
+}
+
+/**
+ * Properties for configuring an Item Reader that iterates over items in a CSV file in S3
+ */
+export interface S3CsvItemReaderProps extends S3ItemReaderProps {
+  /**
+   * CSV file header configuration
+   *
+   * @default - CsvHeaders with CsvHeadersLocation.FIRST_ROW
+   */
+  readonly csvHeaders?: CsvHeaders;
+}
+
+/**
+ * Item Reader configuration for iterating over items in a CSV file stored in S3
+ */
+export class S3CsvItemReader extends S3ItemReader {
+  /**
+   * CSV headers configuration
+   */
+  readonly csvHeaders: CsvHeaders;
+  protected readonly inputType: string = 'CSV';
+
+  constructor(props: S3CsvItemReaderProps) {
+    super(props);
+    this.csvHeaders = props.csvHeaders || CsvHeaders.useFirstRow();
```

Comment:
```suggestion
    this.csvHeaders = props.csvHeaders ?? CsvHeaders.useFirstRow();
```

Anywhere else too. Use nullish coalescing